From a1cef1d254b08719d9ac2e90215f6e33795839c5 Mon Sep 17 00:00:00 2001
From: Artem Senichev <a.senichev@yadro.com>
Date: Thu, 12 Jul 2018 11:50:54 +0300
Subject: [PATCH] Disable set the iv_needsReset for OCC

Hank CH Chang <chhank@tw.ibm.com> wrote:
We think the root cause of throttling should be the first error that
opal-prd/htmgt finds -- which is the abnormal occ reset caused by the
occ present bit conflicts.
Now we're still looking for a general patch/fix for this 4-Socket OCC
present bit issue from our internal hostboot/occ team.
Meanwhile, you could keep the quick fix as a workaround due to we know
from the results -- no throttling, the occ present byte and status looks
good. And traced the log & code you would find that the related
hostboot/htmgt codes just not include the 4 socket situations.
(the updateOccPresentBits error log starts from the 3rd functional
Proc due to the iv_occArray has multi-slave OCCs present in the for-loop,
you might wanna try to modify/update the present code logic
here: https://github.com/open-power/hostboot/blob/master-p8/src/usr/htmgt/htmgt_occ.C#L431)
We find the related htmgt/occ code might not consider well for
the 4 socket situation. So the present bit would conflict in your case:
https://github.com/open-power/hostboot/blob/master-p8/src/usr/htmgt/htmgt_occ.C#L431

Signed-off-by: Artem Senichev <a.senichev@yadro.com>
---
 src/usr/htmgt/htmgt_occ.C | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/usr/htmgt/htmgt_occ.C b/src/usr/htmgt/htmgt_occ.C
index 8b34c525a..7395301dd 100644
--- a/src/usr/htmgt/htmgt_occ.C
+++ b/src/usr/htmgt/htmgt_occ.C
@@ -169,7 +169,6 @@ namespace HTMGT
             TMGT_ERR("updateOccPresentBits: slave 0x%02X already "
                      "exists (0x%02X)",
                      i_slavePresent, iv_occsPresent);
-            iv_needsReset = true;
         }
         else
         {
-- 
2.17.1

